peerconnection.go

```diff
@@ -784,6 +784,13 @@ func (pc *PeerConnection) startReceiver(incoming trackDetails, receiver *RTPRece
 		return
 	}
 
+	// set track id and label early so they can be set as new track information
+	// is received from the SDP.
+	receiver.Track().mu.Lock()
+	receiver.Track().id = incoming.id
+	receiver.Track().label = incoming.label
+	receiver.Track().mu.Unlock()
```

Comment:
I think it is not guaranteed to be executed before the expected timing.
```diff
$ git diff -w
diff --git a/peerconnection.go b/peerconnection.go
index 2c3c69c..d1c6b4c 100644
--- a/peerconnection.go
+++ b/peerconnection.go
@@ -791,6 +791,7 @@ func (pc *PeerConnection) startReceiver(incoming trackDetails, receiver *RTPRece
        receiver.Track().label = incoming.label
        receiver.Track().mu.Unlock()
 
+       go func() {
                if err = receiver.Track().determinePayloadType(); err != nil {
                        pc.log.Warnf("Could not determine PayloadType for SSRC %d", receiver.Track().SSRC())
                        return
@@ -820,6 +821,7 @@ func (pc *PeerConnection) startReceiver(incoming trackDetails, receiver *RTPRece
                } else {
                        pc.log.Warnf("OnTrack unset, unable to handle incoming media streams")
                }
+       }()
 }
 
 // startRTPReceivers opens knows inbound SRTP streams from the RemoteDescription
@@ -858,7 +860,7 @@ func (pc *PeerConnection) startRTPReceivers(incomingTracks map[uint32]trackDetai
 
                        delete(incomingTracks, ssrc)
                        localTransceivers = append(localTransceivers[:i], localTransceivers[i+1:]...)
-                       go pc.startReceiver(incoming, t.Receiver())
+                       pc.startReceiver(incoming, t.Receiver())
                        break
                }
        }
@@ -872,7 +874,7 @@ func (pc *PeerConnection) startRTPReceivers(incomingTracks map[uint32]trackDetai
                                pc.log.Warnf("Could not add transceiver for remote SSRC %d: %s", ssrc, err)
                                continue
                        }
-                       go pc.startReceiver(incoming, t.Receiver())
+                       pc.startReceiver(incoming, t.Receiver())
                }
        }
 }
@@ -960,7 +962,7 @@ func (pc *PeerConnection) drainSRTP() {
                                        pc.log.Warnf("Could not add transceiver for remote SSRC %d: %s", ssrc, err)
                                        return false
                                }
-                               go pc.startReceiver(incoming, t.Receiver())
+                               pc.startReceiver(incoming, t.Receiver())
                                return true
                        }
                }
```
would be safe.